core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5080,10 +5081,20 @@ private static void computeVersion(ServletContext context) {
     @Restricted(NoExternalUse.class)
     public static final String UNCOMPUTED_VERSION = "?";
 
+    private static String VERSION = UNCOMPUTED_VERSION;
+
     /**
      * Version number of this Jenkins.
      */
-    public static String VERSION = UNCOMPUTED_VERSION;
+    @AdaptField(name = "VERSION", was = String.class)
+    public static @CheckForNull String getJenkinsVersion() {
+        Jenkins jenkins = getInstanceOrNull();
+        return jenkins != null && jenkins.hasPermission(READ) ? VERSION : null;
+    }
+
+    public static void setJenkinsVersion(String version) {
```

Comment:
setting the version always seemed like an accident waiting to happen.  do we actually need it?